build.sbt

```diff
@@ -1,120 +1,110 @@
-  "org.scalatest"                %% "scalatest"                     % "3.0.5" % "it, test",
-  "com.stephenn"                 %% "scalatest-json-jsonassert"     % "0.0.3" % "it, test",
-  "org.scalamock"                %% "scalamock"                     % "4.1.0" % "it, test",
-  "org.testcontainers"           % "postgresql"                     % "1.12.3" % "it, test",
-  "com.softwaremill.diffx"       %% "diffx-scalatest"               % "0.3.3" % "it, test",
-  "org.http4s"                   %% "http4s-blaze-client"           % http4sVersion % IntegrationTest,
-  "org.http4s"                   %% "http4s-dsl"                    % http4sVersion % IntegrationTest,
-  "org.http4s"                   %% "http4s-circe"                  % http4sVersion % IntegrationTest,
-  compilerPlugin("com.github.ghik" % "silencer-plugin" % silencerVersion cross CrossVersion.full),
-  "com.github.ghik" % "silencer-lib" % silencerVersion % Provided cross CrossVersion.full
-)
+lazy val conseil = (project in file("."))
+  .disableAssembly()
+  .aggregate(common, api, lorre, lorreSchema, smokeTests)
 
-excludeDependencies ++= Seq(
-  "org.consensusresearch" %% "scrypto"
-)
+lazy val common = (project in file("conseil-common"))
+  .settings(
+    name := "conseil-common",
+    libraryDependencies ++= Dependencies.conseilCommonInclude,
+    excludeDependencies ++= Dependencies.conseilCommonExclude,
+    coverageExcludedPackages := Seq(
+          "<empty>",
+          "tech.cryptonomic.conseil.common.io.*",
+          "tech.cryptonomic.conseil.common.tezos.Tables",
+          "tech.cryptonomic.conseil.common.config.Security"
+        ).mkString(";")
+  )
+  .settings(Defaults.itSettings)
+  .settings(
+    scalacOptions += "-P:silencer:pathFilters=common/src/main/scala/tech/cryptonomic/conseil/common/tezos/Tables.scala"
+  )
+  .configs(IntegrationTest)
+  .enableBuildInfo()
+  .disableAssembly()
 
-assemblyOutputPath in assembly := file("/tmp/conseil.jar")
+lazy val api = (project in file("conseil-api"))
+  .settings(
+    name := "conseil-api",
+    mainClass := Some("tech.cryptonomic.conseil.api.Conseil"),
+    libraryDependencies ++= Dependencies.conseilApiInclude,
+    coverageExcludedPackages := Seq(
+          "<empty>",
+          ".*\\.Conseil",
+          ".*\\.ConseilAppConfig"
+        ).mkString(";")
+  )
+  .settings(Defaults.itSettings)
+  .configs(IntegrationTest)
+  .addRunCommand(
+    description = "A run application Task.",
+    javaExtras = Seq("-Xms1024M", "-Xmx8192M", "-Xss1M", "-XX:+CMSClassUnloadingEnabled")
+  )
+  .enableAssembly()
+  .dependsOn(common % "compile->test")
 
-scalacOptions ++= ScalacOptions.common
-scalacOptions += "-P:silencer:pathFilters=src/main/scala/tech/cryptonomic/conseil/tezos/Tables.scala"
-import complete.DefaultParsers._
+lazy val lorre = (project in file("conseil-lorre"))
+  .settings(
+    name := "conseil-lorre",
+    mainClass := Some("tech.cryptonomic.conseil.lorre.Lorre"),
```

Comment:
You are talking about renaming package `lorre` -> `indexer`? I left that, because with Conseil-api we have similar story, but we decided to name it `api` at this stage. if we want to rename it to `indexer` we should name this module like that too, to keep consistency, I think.

Reply:
I think I see your point, fine for me to leave it like this.

Simply I didn't like the Lorre packaging since it doesn't convey meaning. Product/project-like names are usually for top level packaging in my head, and that part is already reserved for `conseil`.

Reply:
I did changed that, so we have `.indexer.` instead of `Lorre`